src/java.net.http/share/classes/jdk/internal/net/http/Stream.java

```diff
@@ -554,6 +553,12 @@ void incoming_reset(ResetFrame frame) {
         } else {
             Flow.Subscriber<?> subscriber =
                     responseSubscriber == null ? pendingResponseSubscriber : responseSubscriber;
+            if (!requestBodyCF.isDone()) {
+                // If a RST_STREAM is received, complete the requestBody. This will allow the
+                // response to be read before the Reset is handled in the case where the client's
+                // input stream is partially consumed or not consumed at all by the server.
+                requestBodyCF.complete(null);
```

Comment:
Doesn't seem right: if the error code != NO_ERROR we should probably complete the requestBodyCf exceptionally?

Reply:
There is now a `completeExceptionally(new IOException("..."))` call is if `frame.ErrorCode() != NO_ERROR`. Otherwise, `complete(null)` is called.

Reply:
Agree that this is better behaviour. Should we test for the case where an ErrorCode is given?